src/main/java/Parser.java

```diff
@@ -0,0 +1,93 @@
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.time.LocalTime.parse;
+
+public class Parser {
+    private CommandType commandType;
+    private boolean isDone;
+    private String description;
+    private String date = "";
+    private LocalDateTime date;
+    private boolean isSafe = true;
```

Comment:
a more descriptive variable name could be used. Quite unclear what this represents at first glance